src/proto_tcp.c

```diff
@@ -690,13 +690,22 @@ int tcp_bind_listener(struct listener *listener, char *errmsg, int errlen)
 		/* we may want to try to restore the default MSS if the socket was inherited */
 		int tmpmaxseg = -1;
 		int defaultmss;
+		int v4 = listener->rx.addr.ss_family == AF_INET;
+		int mptcp = listener->rx.proto->sock_prot == IPPROTO_MPTCP;
 		socklen_t len = sizeof(tmpmaxseg);
 
-		if (listener->rx.addr.ss_family == AF_INET)
-			defaultmss = sock_inet_tcp_maxseg_default;
-		else
-			defaultmss = sock_inet6_tcp_maxseg_default;
-
+		if (mptcp) {
+			if (v4)
+				defaultmss = sock_inet_mptcp_maxseg_default;
+			else
+				defaultmss = sock_inet6_mptcp_maxseg_default;
+		}else{
```

Comment:
This is not following the code-style: `} else {`

Their doc is (too) long: https://github.com/haproxy/haproxy/blob/master/doc/coding-style.txt

But basically, it is very closed to the one from the kernel. You can use the 'checkpatch.pl' script from the kernel as mentioned at the beginning of the document. It should help you to identify issues.

Note that you should also use `--git HEAD` instead of extracting the patch, then `< patch`

Reply:
I did see this script in their doc, and tried to use it, but it didn't show any error. Turns out that the script I downloaded was simply empty because the link they gave in their coding-style was expired...

Reply:
oh, http://git.kernel.org/cgit/linux/kernel/git/torvalds/linux.git/plain/scripts/checkpatch.pl seems to work for me. My browser switches to https by default, maybe that?

Reply:
I did a curl, that's probably the reason then